### PR TITLE
[DRAFT - WIP] Add example for CustomTest command

### PR DESCRIPTION
**Pull request by PriyaModali**
#5383

Create an example for CustomTest command

### CODE REVIEW

File: examples/custom-tests/main.go [hunk 1]
```diff
@@ -0,0 +1,14 @@
+package main
+
+import (
+	"fmt"
+	"time"
+)
+
+func main() {
```
Comment (tejal29) [inline examples/custom-tests/main.go:8]: can this example be a goTest with `TestMain` ?

File: examples/custom-tests/README.md [hunk 2]
```diff
@@ -0,0 +1,25 @@
+### Example: Running custom tests on built images
+
+This example shows how to run
+[custom tests]
+on newly built images in your skaffold dev loop. Tests are associated with single
```
Comment (tejal29) [inline examples/custom-tests/README.md:5]: Please explain more how test dependencies is different than source dependencies.
1) when test file changes, no build will happen but tests would get re-run

**Review by tejal29 — CHANGES_REQUESTED** [review]

File: examples/custom-tests/test/profile_structure_test.yaml [hunk 3]
```diff
@@ -0,0 +1,6 @@
+schemaVersion: 2.0.0
+
+fileExistenceTests:
+  - name: 'no go binary'
```
[the remark this answers is not in the document]
Comment (PriyaModali) [reply examples/custom-tests/test/profile_structure_test.yaml:4]: This is an existing structure test.

File: examples/custom-tests/skaffold.yaml [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+        dependencies:
+          paths:
+          - "**.go"
+    structureTests:
+      - ./test/*
+deploy:
+  kubectl:
+    manifests:
+      - k8s-*
+profiles:
```
[the remark this answers is not in the document]
Comment (PriyaModali) [reply examples/custom-tests/skaffold.yaml:20]: Included `profiles` section:
1. To show example usage in profiles
2. To be consistent with Structure tests

File: examples/custom-tests/main.go (hunk 1 above)
Comment (PriyaModali) [reply examples/custom-tests/main.go:8]: Replaced the existing example with the suggested one.

File: examples/custom-tests/test/main.go [hunk 5]
```diff
@@ -0,0 +1,187 @@
+/*
```
[the remark this answers is not in the document]
Comment (PriyaModali) [reply examples/custom-tests/test/main.go:1]: Removed

File: examples/custom-tests/skaffold.yaml [hunk 6]
```diff
@@ -0,0 +1,31 @@
...
+  artifacts:
+  - image: skaffold-example
+test:
+  - image: skaffold-example
+    custom:
+      - command: ./test/gotest.sh
+        timeout: 300 #in seconds - 5mins
+        dependencies:
+          paths:
+          - "**.go"
```
[the remark this answers is not in the document]
Comment (PriyaModali) [reply examples/custom-tests/skaffold.yaml:13]: Updated the samples.

**Comment by PriyaModali** [comment]
Closing this PR as the custom test example has been added as part of #5451.

**Pull request closed without merging**